aws-s3-transfer-manager/src/operation/download/object_meta.rs

```diff
@@ -116,12 +119,35 @@ impl ObjectMetadata {
                 total.parse().expect("valid range total")
             }
             (Some(length), None) => {
-                debug_assert!(length > 0, "content length invalid");
+                debug_assert!(length >= 0, "content length invalid");
                 length as u64
             },
             (None, None) => panic!("total object size cannot be calculated without either content length or content range headers")
         }
     }
+
+    /// <p>Parse the content-range header to the inclusive range..</p>
+    pub(crate) fn range(&self) -> Option<RangeInclusive<u64>> {
```

Comment:
Range is a user side header https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range. 
```suggestion
    pub(crate) fn parse_range_from_content_range(&self) -> Option<RangeInclusive<u64>> {
```